preprocessy/handlenullvalues/_main.py

```diff
@@ -115,7 +123,13 @@ def __fill_values_columns(self):
             self.new_df[column].fillna(self.fill_values[column], inplace=True)
         return self.new_df
 
-    def execute(self):
+    # main function
+    def execute(self, params):
+
```

Comment:
Add the part from `__validate_input` here - 

```python
def execute(self, params):
    if "df" in params.keys():
         self.df = params["df"]
    .
    .
    .
    self.__validate_input()
```

This will simplify things

Reply:
Okay, will do this